cupyx/fallback_mode/fallback.py

```diff
@@ -88,6 +88,33 @@ def __repr__(self):
     def __doc__(self):
         return self._numpy_object.__doc__
 
+    @staticmethod
+    def _is_cupy_compatible(args, kwargs):
+        """
+        Returns False if ndarray is not compatible with CuPy.
+        """
+        for arg in args:
+            if isinstance(arg, ndarray) and arg._class is not cp.ndarray:
+                return False
+
+        for key in kwargs:
+            if isinstance(kwargs[key], ndarray) and \
+               kwargs[key]._class is not cp.ndarray:
+                return False
+
+        _compatible_dtypes = [
+            np.bool, np.int8, np.int16, np.int32, np.int64,
+            np.uint8, np.uint16, np.uint32, np.uint64,
+            np.float16, np.float32, np.float64,
+            np.complex64, np.complex128
+        ] + list('?bhilqBHILQefdFD')
```

Comment:
Why did you list up twice for each dtype?

Reply:
`numpy.array([1,2,3], dtype=numpy.float64)` OR `numpy.array([1,2,3], dtype='d')`

Reply:
Also `numpy.array(..., dtype='float64')` should be accepted.

Reply:
How about implementing as follows?
```py
_compatible_dtypes = list('?bhilqBHILQefdFD')
dtype = kwargs.get('dtype', None)
if dtype is not None and numpy.dtype(dtype) not in _compatible_dtypes:
    return False
```

Reply:
In fact, dtypes might be passed as `args`. It is difficult to decide that the argument expresses dtype, or not.
How about trying `_call_cupy` once, and fallback to `_call_numpy` if cupy function raised a `NotImplementedError` or a `ValueError`?

Reply:
Yes, though `numpy.dtype(arg).kind` can be used to check an argument for compatible dtype, it will need a try-except block and will probably need to raise an error (to propagate in structure like `_get_xp_args`). So, lets use try-except block in `_RecursiveAttr.__call__` itself. We can then pass flag to `_numpy_call` to indicate CuPy incompatibility.
I think we should except `Exception` for now, `cupy.arange(3, dtype='O')` raises `TypeError`. There could be other such cases.